### PR TITLE
Add revive linter

### PR DESCRIPTION
**Pull request by otherview**
# Description

This PR adds the revive linter to our current golang execution.

It's an extensive change but allows bringing our code to more updated standards of golang formatting.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: .github/workflows/lint-go.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.59.1
+          version: v1.60.3
```
Comment (leszek-vechain) [inline .github/workflows/lint-go.yaml:23]: can we clean up a bit this file - seems like it is throwing errors about: Warning: Unexpected input(s) 'skip-pkg-cache', 'skip-build-cache', valid inputs are ['version', 'install-mode', 'working-directory', 'github-token', 'only-new-issues', 'skip-cache', 'skip-save-cache', 'problem-matchers', 'args', 'cache-invalidation-interval'], probably it is safe to remove these

Comment (otherview) [reply .github/workflows/lint-go.yaml:23]: Where/how are you seeing/reproducing those errors ?

Comment (leszek-vechain) [reply .github/workflows/lint-go.yaml:23]: it is warning that we are using non-existing config options - https://github.com/vechain/thor/actions/runs/10723392301/job/29736573457 just open golangci-lint

Comment (otherview) [reply .github/workflows/lint-go.yaml:23]: I think that was a fluke, I've reran that action and it worked ok.

**Review by leszek-vechain — APPROVED** [review]
I think it looks fine (as far as I understand changes)

File: poa/candidates.go [hunk 2]
```diff
@@ -44,8 +44,8 @@ func NewCandidates(list []*authority.Candidate) *Candidates {
 // Copy make a copy.
 func (c *Candidates) Copy() *Candidates {
 	c.referenced = true
-	copy := *c
-	return &copy
+	cpy := *c
```
Comment (libotony) [inline poa/candidates.go:47]: What's the rule behind this?

File: cmd/thor/main.go [hunk 3]
```diff
@@ -171,22 +171,22 @@ func defaultAction(ctx *cli.Context) error {
 	metricsURL := ""
 	if ctx.Bool(enableMetricsFlag.Name) {
 		metrics.InitializePrometheusMetrics()
-		url, close, err := api.StartMetricsServer(ctx.String(metricsAddrFlag.Name))
+		url, closeFun, err := api.StartMetricsServer(ctx.String(metricsAddrFlag.Name))
```
Comment (libotony) [inline cmd/thor/main.go:174]: `closeFunc` sounds better?

File: p2psrv/server.go [hunk 4]
```diff
@@ -356,11 +356,11 @@ func (s *Server) fetchBootstrap() {
 	}
 
 	for {
-		if err := f(); err == nil || errors.Is(err, context.Canceled) {
+		err := f()
+		if err == nil || errors.Is(err, context.Canceled) {
```
Comment (libotony) [inline p2psrv/server.go:360]:
```suggestion
		if err := f(); err == nil || errors.Is(err, context.Canceled) {
```

File: poa/candidates.go (hunk 2 above)
Comment (otherview) [reply poa/candidates.go:47]: No rule per say, but copy matches the `inbuilt.copy` function.

File: cmd/thor/main.go (hunk 3 above)
Comment (otherview) [reply cmd/thor/main.go:174]: Fixed 👍

File: p2psrv/server.go (hunk 4 above)
Comment (libotony) [reply p2psrv/server.go:360]: Right, didn't see err was reference by the logger

File: poa/candidates.go (hunk 2 above)
Comment (libotony) [reply poa/candidates.go:47]: Isn't that strange that it prefers `cpy` over `copy`?

Comment (libotony) [reply poa/candidates.go:47]: Hmm, `copy` is a reserved keyword :-)

File: cmd/thor/optimizer/optimizer_test.go [hunk 5]
```diff
@@ -134,7 +134,7 @@ func newTempFileDB() (*muxdb.MuxDB, func() error, error) {
 		return nil, nil, err
 	}
 
-	close := func() error {
+	closeFun := func() error {
```
Comment (libotony) [inline cmd/thor/optimizer/optimizer_test.go:137]:
```suggestion
	closeFunc := func() error {
```

File: cmd/thor/optimizer/optimizer_test.go [hunk 6]
```diff
@@ -146,7 +146,7 @@ func newTempFileDB() (*muxdb.MuxDB, func() error, error) {
 		return nil
 	}
 
-	return db, close, nil
+	return db, closeFun, nil
```
Comment (libotony) [inline cmd/thor/optimizer/optimizer_test.go:149]:
```suggestion
	return db, closeFunc, nil
```

Minor, but nice to have :-)

**Review by libotony — APPROVED** [review]

**Pull request merged**